src/lib/ProductExporter.ts

```diff
@@ -477,26 +477,31 @@ export class ProductExporter {
             ...chargeElements,
             ...chargeTiers
         ]);
-      
         for(let charge of charges){
-
-            let chargeName = charge['Name'];
-            let chargeToSave: any = {};
-            let chargeElementsToSave: any = [];
-            let chargeTiersToSave: any = [];
-            chargeToSave.root= charge;
             
-            chargeElements.filter(chargeElement => chargeElement['enxCPQ__Charge_Parent__r'] 
-                                                   && chargeElement['enxCPQ__Charge_Parent__r']['enxCPQ__TECH_External_Id__c'] === charge['enxCPQ__TECH_External_Id__c'])
-                          .forEach(chargeElementToSave=>{ chargeElementsToSave.push(chargeElementToSave)});
-
-            chargeTiers.filter(chargeTier => chargeTier['enxCPQ__Charge_Parent__r']
-                                             && chargeTier['enxCPQ__Charge_Parent__r']['enxCPQ__TECH_External_Id__c'] === charge['enxCPQ__TECH_External_Id__c'])
-                       .forEach(chargeTierToSave=>{ chargeTiersToSave.push(chargeTierToSave)});              
-
-            chargeToSave.chargeElements = chargeElementsToSave;
-            chargeToSave.chargeTier = chargeTiersToSave;
-            Util.writeFile('/charges/' + Util.sanitizeFileName(chargeName) + '.json', chargeToSave);
+            let chargeName = charge['Name'];
+            let chargeTechId = charge['enxCPQ__TECH_External_Id__c'];
+            let chargeFileName = chargeName + chargeTechId;
+
+            if(chargeFileNames.has(chargeFileName)){
+                let chargeToSave: any = {};
+                let chargeElementsToSave: any = [];
+                let chargeTiersToSave: any = [];
+                chargeToSave.root= charge;
+                
+                chargeElements.filter(chargeElement => chargeElement['enxCPQ__Charge_Parent__r'] 
+                                                       && chargeElement['enxCPQ__Charge_Parent__r']['enxCPQ__TECH_External_Id__c'] === charge['enxCPQ__TECH_External_Id__c'])
+                              .forEach(chargeElementToSave=>{ chargeElementsToSave.push(chargeElementToSave)});
+
+                chargeTiers.filter(chargeTier => chargeTier['enxCPQ__Charge_Parent__r']
+                                                 && chargeTier['enxCPQ__Charge_Parent__r']['enxCPQ__TECH_External_Id__c'] === charge['enxCPQ__TECH_External_Id__c'])
+                           .forEach(chargeTierToSave=>{ chargeTiersToSave.push(chargeTierToSave)});              
+
+                chargeToSave.chargeElements = chargeElementsToSave;
+                chargeToSave.chargeTier = chargeTiersToSave;
```

Comment:
You could do this in simmilar way but using map - in that case you don't need can skip couple of lines

Reply:
Also I'd probably extract that to another method